python/ray/rllib/agents/ddpg/ddpg_policy_graph.py

```diff
@@ -189,13 +239,16 @@ def __init__(self, observation_space, action_space, config):
 
         # Action outputs
         with tf.variable_scope(A_SCOPE, reuse=True):
-            deterministic_flag = tf.constant(value=False, dtype=tf.bool)
+            deterministic_flag = tf.constant(value=True, dtype=tf.bool)
             zero_eps = tf.constant(value=.0, dtype=tf.float32)
             output_actions = self._build_action_network(
                 self.p_t, deterministic_flag, zero_eps)
-
             output_actions_estimated = self._build_action_network(
-                p_tp1, deterministic_flag, zero_eps)
+                p_tp1,
+                tf.constant(value=False, dtype=tf.bool)
+                if self.config["smooth_target_policy"] else deterministic_flag,
+                zero_eps,
```

Comment:
Similarly I think you can just pass 0.0 for zero_eps.

Reply:
like the above, if we also consider eps for Gaussian, eps can be 0.0 for output_action, but can NOT be  0.0 for output_action_estimation (i.e., the target action).